docs/custom_quantum_functions.md

```diff
@@ -156,11 +172,47 @@ void matrix_init_user(void) {
 }
 ```
 
+### `keyboard_pre_init_*` Function Documentation
+
+* Keyboard/Revision: `void keyboard_pre_init_kb(void)`
+* Keymap: `void keyboard_pre_init_user(void)`
+
+## Matrix Initialization Code
+
+This is called when the matrix is initialized, and after some of the hardware has been set up, but before many of the features have been initialized. 
+
+This is useful for setting up stuff that you may need elsewhere, but isn't hardware related nor is dependant on where it's started. 
+
+
 ### `matrix_init_*` Function Documentation
 
 * Keyboard/Revision: `void matrix_init_kb(void)`
 * Keymap: `void matrix_init_user(void)`
 
+
+## Keyboard Post Initialization code
+
+This is ran as the very last task in the keyboard initialization process. This is useful if you want to make changes to certain features, as they should be initialized by this point.
+
+
+### Example `keyboard_post_init_user()` Implementation
+
+This example, running after everything else has initialized, sets up the rgb underglow configuration.
```

Comment:
Consider being more explicit somewhere about _this_ being where you do rgb init stuff, as the other init functions run too early.

Reply:
THis is, actually.  Check line 148:
```md
!> For most people, the `keyboard_post_init_user` function is what you want to call.  For instance, this is where you want to set up things for RGB Underglow.
```

Because, yeah, it needs to be very visible and obvious.

Reply:
Ah yeah, ok. I assume the `>!` shows up more obviously on the docs, but not on github markdown?

Reply:
Correct.  

https://docs.qmk.fm/#/documentation_best_practices?id=important

Reply:
So, if that's not obvious enough....

Reply:
No, that's fine, it just didn't stick out much in the plain text version.

Reply:
But I can't mark conversations as resolved, even when they're my own :)